From 4d7277f72c8f4915f237e36982ffd7dfba524b15 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Mon, 7 Aug 2023 18:00:34 +0800
Subject: [PATCH] print: Revert "Start sorting apart includes" change for
 gtkprinteroptionprivate.h

The print backends needs the functions in gtkprinteroptionprivate.h to
be exported.

Fixes #6003.
---
 gtk/print/gtkprinteroptionprivate.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/gtk/print/gtkprinteroptionprivate.h b/gtk/print/gtkprinteroptionprivate.h
index 82dd8739ba1..a7d8e8ad5b6 100644
--- a/gtk/print/gtkprinteroptionprivate.h
+++ b/gtk/print/gtkprinteroptionprivate.h
@@ -27,6 +27,7 @@
 #endif
 
 #include <glib-object.h>
+#include <gdk/gdk.h>
 
 G_BEGIN_DECLS
 
@@ -87,28 +88,39 @@ struct _GtkPrinterOptionClass
   void (*_gtk_reserved4) (void);
 };
 
+GDK_AVAILABLE_IN_ALL
 GType   gtk_printer_option_get_type       (void) G_GNUC_CONST;
 
+GDK_AVAILABLE_IN_ALL
 GtkPrinterOption *gtk_printer_option_new                    (const char           *name,
 							     const char           *display_text,
 							     GtkPrinterOptionType  type);
+GDK_AVAILABLE_IN_ALL
 void              gtk_printer_option_set                    (GtkPrinterOption     *option,
 							     const char           *value);
+GDK_AVAILABLE_IN_ALL
 void              gtk_printer_option_set_has_conflict       (GtkPrinterOption     *option,
 							     gboolean              has_conflict);
+GDK_AVAILABLE_IN_ALL
 void              gtk_printer_option_clear_has_conflict     (GtkPrinterOption     *option);
+GDK_AVAILABLE_IN_ALL
 void              gtk_printer_option_set_boolean            (GtkPrinterOption     *option,
 							     gboolean              value);
+GDK_AVAILABLE_IN_ALL
 void              gtk_printer_option_allocate_choices       (GtkPrinterOption     *option,
 							     int                   num);
+GDK_AVAILABLE_IN_ALL
 void              gtk_printer_option_choices_from_array     (GtkPrinterOption     *option,
                                                              int                   num_choices,
                                                              const char           **choices,
                                                              const char           **choices_display);
+GDK_AVAILABLE_IN_ALL
 gboolean          gtk_printer_option_has_choice             (GtkPrinterOption     *option,
 							    const char           *choice);
+GDK_AVAILABLE_IN_ALL
 void              gtk_printer_option_set_activates_default (GtkPrinterOption     *option,
 							    gboolean              activates);
+GDK_AVAILABLE_IN_ALL
 gboolean          gtk_printer_option_get_activates_default (GtkPrinterOption     *option);
 
 
-- 
GitLab

